File: data/Pandora.Data/Transformers/TerraformMethodDefinition.cs
using Pandora.Data.Helpers;

namespace Pandora.Data.Transformers;

public static class TerraformMethodDefinition
{
    public static Models.TerraformMethodDefinition Map(Definitions.Interfaces.MethodDefinition input)
    {
        var methodName = input.Method.Name.TrimSuffix("Operation");
        return new Models.TerraformMethodDefinition
        {
            Generate = input.Generate,
            MethodName = methodName,
            TimeoutInMinutes = input.TimeoutInMinutes,
        };
    }
}

File: tools/data-api-repository/transforms/terraform_method_definition.go
// Copyright (c) HashiCorp, Inc.
// SPDX-License-Identifier: MPL-2.0

package transforms

import (
	sdkModels "github.com/hashicorp/pandora/tools/data-api-sdk/v1/models"
	repositoryModels "github.com/hashicorp/pandora/tools/sdk/dataapimodels"
)

func mapTerraformMethodDefinitionToRepository(input sdkModels.TerraformMethodDefinition) repositoryModels.TerraformMethodDefinition {
	return repositoryModels.TerraformMethodDefinition{
		Generate:         input.Generate,
		Name:             input.SDKOperationName,
		TimeoutInMinutes: input.TimeoutInMinutes,
	}
}
